charmcraft/commands/store/__init__.py

```diff
@@ -769,15 +769,25 @@ def fill_parser(self, parser):
     def run(self, parsed_args):
         """Run the command."""
         store = Store(self.config.charmhub)
+        packages_list = store.list_registered_names(include_collaborations=True)
+        packages_type_map = {
+            p.name: p.entity_type for p in packages_list if p.name == parsed_args.name
+        }
```

Comment:
We could obtain the package type directly without using a map (which now would have only one entry?)

Reply:
Updated.